src/merkle/store/tests.rs

```diff
@@ -647,6 +647,149 @@ fn test_constructors() -> Result<(), MerkleError> {
+
+    // flip the key to the right of the root and insert some content on depth 16
+    let key = 0b11001011_10110110_00000000_00000000_00000000_00000000_00000000_00000000_u64;
+    assert_eq!(1, store.get_leaf_depth(root, key).unwrap());
+    let depth = 16;
+    let index = NodeIndex::new(depth, key >> (64 - depth)).unwrap();
+    let node = [Felt::new(key); WORD_SIZE];
+    root = store.set_node(root, index, node).unwrap().root;
+    assert_eq!(depth, store.get_leaf_depth(root, key).unwrap());
+
+    // attempt the sibling of the previous leaf
+    let key = 0b11001011_10110111_00000000_00000000_00000000_00000000_00000000_00000000_u64;
+    assert_eq!(16, store.get_leaf_depth(root, key).unwrap());
+    let index = NodeIndex::new(depth, key >> (64 - depth)).unwrap();
+    let node = [Felt::new(key); WORD_SIZE];
+    root = store.set_node(root, index, node).unwrap().root;
+    assert_eq!(depth, store.get_leaf_depth(root, key).unwrap());
+
+    // move down to the next depth and assert correct behavior
+    let key = 0b11001011_10110100_00000000_00000000_00000000_00000000_00000000_00000000_u64;
+    assert_eq!(15, store.get_leaf_depth(root, key).unwrap());
+    let depth = 17;
+    let index = NodeIndex::new(depth, key >> (64 - depth)).unwrap();
+    let node = [Felt::new(key); WORD_SIZE];
+    root = store.set_node(root, index, node).unwrap().root;
+    assert_eq!(depth, store.get_leaf_depth(root, key).unwrap());
+}
+
+#[test]
+fn get_leaf_depth_works_with_small_depth() {
+    let mut store = MerkleStore::new();
+    let mut root: Word = EmptySubtreeRoots::empty_hashes(64)[0].into();
+    let node = [ONE; WORD_SIZE];
+
+    // insert some random, 8 depth keys. `a` diverges from the first bit
+    let a = 0b01101001_00000000_00000000_00000000_00000000_00000000_00000000_00000000_u64;
+    let b = 0b10011001_00000000_00000000_00000000_00000000_00000000_00000000_00000000_u64;
+    let c = 0b10010110_00000000_00000000_00000000_00000000_00000000_00000000_00000000_u64;
+    let d = 0b11110110_00000000_00000000_00000000_00000000_00000000_00000000_00000000_u64;
+
+    for k in [a, b, c, d] {
+        let index = NodeIndex::new(8, k >> 56).unwrap();
+        root = store.set_node(root, index, node).unwrap().root;
+    }
+
+    // assert all leaves returns the inserted depth
+    for k in [a, b, c, d] {
+        assert_eq!(8, store.get_leaf_depth(root, k).unwrap());
+    }
+
+    // flip last bit of a and expect it to return the the same depth, but for an empty node
+    assert_eq!(8, store.get_leaf_depth(root, a ^ (1 << 56)).unwrap());
+
+    // flip fourth bit of a and expect an empty node on depth 4
+    assert_eq!(4, store.get_leaf_depth(root, a ^ (1 << 60)).unwrap());
+
+    // flip third bit of a and expect an empty node on depth 3
+    assert_eq!(3, store.get_leaf_depth(root, a ^ (1 << 61)).unwrap());
+
+    // flip second bit of a and expect an empty node on depth 2
+    assert_eq!(2, store.get_leaf_depth(root, a ^ (1 << 62)).unwrap());
```

Comment:
I think the keys are simple enough where we could just do something like
```Rust
// flip last bit of a and expect it to return the the same depth, but for an empty node
assert_eq!(8, store.get_leaf_depth(root, 0b0110_1000).unwrap());

// flip fourth bit of a and expect an empty node on depth 4
assert_eq!(4, store.get_leaf_depth(root, 0b0111).unwrap());

// flip second bit of a and expect an empty node on depth 2
assert_eq!(2, store.get_leaf_depth(root, 0b00).unwrap());
```

Reply:
We are using the first depths there, so the result probably woudn't be `8`. Let's centralize it here:

https://github.com/0xPolygonMiden/crypto/pull/125#discussion_r1159045463